frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/k8s-ref.ts

```diff
@@ -0,0 +1,77 @@
+import { K8sModel } from '../../api/common-types';
+import {
+  K8sResourceKindReference,
+  K8sGroupVersionKind,
+  K8sResourceCommon,
+} from '../../extensions/console-types';
+
+/**
+ * Provides a reference string that uniquely identifies the group, version, and kind of a k8s resource.
+ * @param group Pass group of k8s resource or model.
+ * @param version Pass version of k8s resource or model.
+ * @param kind Pass kind of k8s resource or model.
+ * @return The reference for any k8s resource i.e `group~version~kind`.
+ * * */
+export const getReference = (
+  group: string,
+  version: string,
+  kind: string,
+): K8sResourceKindReference => [group, version, kind].join('~');
+
+/**
+ * Provides a reference string that uniquely identifies the group, version, and kind of a k8s model.
+ * @param model k8s model
+ * @return The reference for model i.e `group~version~kind`.
+ * * */
+export const getReferenceForModel = (model: K8sModel): K8sResourceKindReference =>
+  getReference(model.apiGroup || 'core', model.apiVersion, model.kind);
```

Comment:
If you make the suggested change to `getReference`, this could be

```suggestion
  getReference(model.apiGroup, model.apiVersion, model.kind);
```